src/gsy_e/models/strategy/external_strategies/__init__.py

```diff
@@ -316,6 +320,117 @@ def _get_market_from_time_slot(self, time_slot: DateTime) -> MarketBase:
                 f"settlement markets")
         return market
 
+    def _offer_aggregator_impl(
+            self, arguments: Dict, market: "MarketBase", time_slot: DateTime,
+            available_energy: float) -> Dict:
+        """Post offer in the market for aggregator connection to load or PV."""
+        response_message = ""
+        arguments, filtered_fields = self.filter_degrees_of_freedom_arguments(arguments)
+        if filtered_fields:
+            response_message = (
+                "The following arguments are not supported for this market and have been "
+                f"removed from your order: {filtered_fields}.")
+
+        required_args = {"price", "energy", "type", "transaction_id"}
+        allowed_args = required_args.union({"replace_existing",
+                                            "time_slot",
+                                            "attributes",
+                                            "requirements"})
+        try:
+            # Check that all required arguments have been provided
+            assert all(arg in arguments.keys() for arg in required_args)
+            # Check that every provided argument is allowed
+            assert all(arg in allowed_args for arg in arguments.keys())
+
+            replace_existing = arguments.pop("replace_existing", True)
+
+            assert self.can_offer_be_posted(
+                arguments["energy"],
+                arguments["price"],
+                available_energy,
+                market,
+                time_slot=time_slot,
+                replace_existing=replace_existing)
+
+            offer_arguments = {k: v
+                               for k, v in arguments.items()
+                               if k not in ["transaction_id", "type", "time_slot"]}
+
+            offer = self.post_offer(
+                market, replace_existing=replace_existing, **offer_arguments)
+
+            response = {
+                "command": "offer",
+                "status": "ready",
+                "offer": offer.to_json_string(replace_existing=replace_existing),
+                "transaction_id": arguments.get("transaction_id"),
+                "area_uuid": self.device.uuid,
+                "message": response_message}
+        except (AssertionError, GSyException):
+            logging.exception("Failed to post PV offer.")
```

Comment:
Please remove the PV reference since it is on the ExternalMixin which has no knowledge of the strategy type. Alternatively, you can do the following `f"Failed to post offer on strategy {self.__class__.__name__}"` to get the class name and have the indication which strategy was affected.

Reply:
Ah yes, thanks!